### PR TITLE
Add type aliases for ElasticVector and ElasticMatrix

### LINKED ISSUE #27: Add type aliases for Vector and Matrix?
Julia has aliases for 1D and 2D arrays named `Vector` and `Matrix` respectively.  I feel it is convenient to use them to indicate explicitly that we are dealing with a 1D or 2D arrays, e.g., in a function interface.  Is it a good idea to supply such aliases in `ElasticArrays` for consistency? 
E.g., 
```julia
const ElasticVector{T} = ElasticArray{T, 1}
const ElasticMatrix{T} = ElasticArray{T, 2}
```

### PR DESCRIPTION
**Pull request by ShuhuaGao**
Fixes #27

### CODE REVIEW

File: src/elasticarray.jl [hunk 1]
```diff
@@ -19,7 +19,7 @@ Constructors:
     ElasticArray{T}(dims::Integer...)
     convert(ElasticArray, A::AbstractArray)
 """
-struct ElasticArray{T,N,M,V<:DenseVector{T}} <: DenseArray{T,N}
+struct ElasticArray{T,N,M,V <: DenseVector{T}} <: DenseArray{T,N}
```
Comment (oschulz) [inline src/elasticarray.jl:22]: Could you maybe remove these whitespace changes?

File: src/elasticarray.jl [hunk 2]
```diff
@@ -64,7 +64,7 @@ ElasticArray{T,N}(A::AbstractArray{<:Any,N}) where {T,N} = copyto!(ElasticArray{
 ElasticArray{T}(A::AbstractArray{<:Any,N}) where {T,N} = copyto!(ElasticArray{T,N}(undef, size(A)), A)
 ElasticArray(A::AbstractArray{T,N}) where {T,N} = copyto!(ElasticArray{T,N}(undef, size(A)), A)
 
-Base.convert(::Type{T}, A::AbstractArray) where {T<:ElasticArray} = A isa T ? A : T(A)
+Base.convert(::Type{T}, A::AbstractArray) where {T <: ElasticArray} = A isa T ? A : T(A)
```
Comment (oschulz) [inline src/elasticarray.jl:67]: Could you maybe remove these whitespace changes?

File: src/elasticarray.jl [hunk 3]
```diff
@@ -50,7 +50,7 @@ function ElasticArray{T,N,M,V}(::UndefInitializer, dims::NTuple{N,Integer}) wher
     ElasticArray{T,N,M,V}(kernel_size, SignedMultiplicativeInverse{Int}(kernel_length), data)
 end
 ElasticArray{T,N,M}(::UndefInitializer, dims::NTuple{N,Integer}) where {T,N,M} = ElasticArray{T,N,M,Vector{T}}(undef, dims)
-ElasticArray{T,N}(::UndefInitializer, dims::NTuple{N,Integer}) where {T,N} = ElasticArray{T,N,N-1}(undef, dims)
+ElasticArray{T,N}(::UndefInitializer, dims::NTuple{N,Integer}) where {T,N} = ElasticArray{T,N,N - 1}(undef, dims)
```
Comment (oschulz) [inline src/elasticarray.jl:53]: Could you maybe remove these whitespace changes?

File: src/elasticarray.jl [hunk 4]
```diff
@@ -229,3 +229,12 @@ function Adapt.adapt_structure(to, A::ElasticArray{<:Any,N,M}) where {N,M}
     data = adapt(to, A.data)
     ElasticArray{eltype(data),N,M,typeof(data)}(A.kernel_size, A.kernel_length, data)
 end
+
+"""
+Type alias for `ElasticArray{T,1}`.
+"""
+const ElasticVector{T} = ElasticArray{T,1}
```
Comment (oschulz) [inline src/elasticarray.jl:236]: There should be some tests for the ctor's, just to make sure all is Ok.

File: src/elasticarray.jl [hunk 5]
```diff
@@ -229,3 +229,12 @@ function Adapt.adapt_structure(to, A::ElasticArray{<:Any,N,M}) where {N,M}
...
+"""
+Type alias for `ElasticArray{T,2}`.
+"""
+const ElasticMatrix{T} = ElasticArray{T,2}
```
Comment (oschulz) [inline src/elasticarray.jl:240]: There should be some tests for the ctor's, just to make sure all is Ok.

**Review by oschulz — CHANGES_REQUESTED** [review]

File: src/elasticarray.jl (hunk 1 above)
Comment (ShuhuaGao) [reply src/elasticarray.jl:22]: Sure. It seems that VS code formats it automatically though. I am not familiar with the PR workflow. Shall I propose a new PR when proper changes have been made?

Comment (oschulz) [reply src/elasticarray.jl:22]: Normally, you can add more commits, but not if you did a PR based on your master branch. In general, the recommendation is to always create a dedicated branch on your clone of the repo for the PR (see https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests).

It's probably best if you close this PR, create a branch, and create a new one.

**Pull request closed without merging**